### PR TITLE
bugfix: take has wrong output type.

### PR DESCRIPTION
**Pull request by benikm91**
While working with DimWit I ran into the following issue:

If we "take" not the first dimension but later dimensions the returned type signatures is wrong. It adds the new dimension always at location 0 rather than the location where the axis is taken (like jnp.take). This PR fixes this issue.

### CODE REVIEW

**Review by marcelluethi — APPROVED** [review]
Thanks for fixing this bug. Looks good to me.
